internal/vault/certificates.go

```diff
@@ -26,13 +26,18 @@ const (
 	defaultIssuer = "default"
 )
 
+//go:generate mockgen -source ./certificates.go -destination ./mocks/certificates.go -package mocks LogicalClient
+type LogicalClient interface {
```

Comment:
I think the `Logical` naming is a bit awkward/confusing, particularly the `.Logical()` getter, could this be `VaultClient` or something instead?

Reply:
I'm ok with the `LogicalClient`, specifically because it makes it awkward having to nest two interfaces for a call to `Logical`/`WriteWithContext`

Reply:
I initially went with VaultClient, but it felt kinda gross since you either nest two interface like Andrew said, or you wind up labeling a very specific set of the Vault API/SDK as `VaultClient` and assuming we'll never need the rest of the API/SDK outside of `vault.NewClient(...).Logical()`. I'm not super opinionated either way though 🤷

Reply:
also the `vault.VaultClient` interface would stutter more than I like 😅